backend/controller/controller.go

```diff
@@ -1810,54 +1811,19 @@ func (s *Service) syncRoutes(ctx context.Context) (ret time.Duration, err error)
 				}
 			}
 			newRoutes[v.Module] = Route{Module: v.Module, Deployment: v.Key, Endpoint: targetEndpoint}
+			modulesByName[v.Module] = v.Schema
 		}
 	}
-	s.routes.Store(newRoutes)
-	return time.Second, nil
-}
-
-// Synchronises Service.schema from the database.
-func (s *Service) syncSchema(ctx context.Context) {
-	logger := log.FromContext(ctx)
-	modulesByName := map[string]*schema.Module{}
-	retry := backoff.Backoff{Max: time.Second * 5}
-	for {
-		err := s.watchModuleChanges(ctx, func(response *ftlv1.PullSchemaResponse) error {
-			switch response.ChangeType {
-			case ftlv1.DeploymentChangeType_DEPLOYMENT_ADDED, ftlv1.DeploymentChangeType_DEPLOYMENT_CHANGED:
-				moduleSchema, err := schema.ModuleFromProto(response.Schema)
-				if err != nil {
-					return err
-				}
-				modulesByName[moduleSchema.Name] = moduleSchema
 
-			case ftlv1.DeploymentChangeType_DEPLOYMENT_REMOVED:
-				delete(modulesByName, response.ModuleName)
-			}
+	orderedModules := maps.Values(modulesByName)
+	sort.SliceStable(orderedModules, func(i, j int) bool {
+		return orderedModules[i].Name < orderedModules[j].Name
+	})
+	combined := &schema.Schema{Modules: orderedModules}
+	s.schema.Store(ftlreflect.DeepCopy(combined))
 
-			orderedModules := maps.Values(modulesByName)
-			sort.SliceStable(orderedModules, func(i, j int) bool {
-				return orderedModules[i].Name < orderedModules[j].Name
-			})
-			combined := &schema.Schema{Modules: orderedModules}
-			s.schema.Store(ftlreflect.DeepCopy(combined))
-			return nil
-		})
-		if err != nil {
-			next := retry.Duration()
-			if ctx.Err() == nil {
-				// Don't log when the context is done
-				logger.Warnf("Failed to watch module changes, retrying in %s: %s", next, err)
-			}
-			select {
-			case <-time.After(next):
-			case <-ctx.Done():
-				return
-			}
-		} else {
-			retry.Reset()
-		}
-	}
+	s.routes.Store(newRoutes)
```

Comment:
To fix the lack of atomicity we could store both in a single atomic.Value as a single struct.

Reply:
In practice I don't think it would matter, however the lack of atomicity just feels wrong. I will move it into a single struct.